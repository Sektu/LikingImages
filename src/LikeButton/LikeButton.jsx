import CustomizedButton from "../CustomizedButton/CustomizedButton";

const LikeButton = (onClick) => {
    return <CustomizedButton
        className="button like-button"
        aria-label="do-like"
        onClick={onClick}
        name="Like">
    </CustomizedButton>;
}

export default LikeButton;