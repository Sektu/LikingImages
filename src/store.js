import { configureStore } from "@reduxjs/toolkit";
import likingReducer from "./slices/likingSlice";
import imagesReducer from "./slices/imagesSlice";

export default configureStore({
  reducer: {
    liking: likingReducer,
    images: imagesReducer,
  }
});
