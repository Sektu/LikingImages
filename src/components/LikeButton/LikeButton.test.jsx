import React from 'react';
import { render, cleanup } from '@testing-library/react';
import store from "../../store";
import { Provider } from "react-redux";
import LikeButton from './LikeButton';

afterEach(cleanup);

it("renders", () => {
    const { asFragment } = render(
        <Provider store={store}>
            <LikeButton onClick={() => {}} />
        </Provider>);
    expect(asFragment()).toMatchSnapshot();
});

it("check like button ", () => {
    const { getByRole } = render(
        <Provider store={store}>
            <LikeButton onClick={() => {}} />
        </Provider>);
    const button = getByRole("button");
    expect(button).toHaveClass("button");
    expect(button).toHaveClass("like-button");
    expect(button).toHaveTextContent("Like");
});
