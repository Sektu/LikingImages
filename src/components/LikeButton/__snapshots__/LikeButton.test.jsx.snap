// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <button
    class="button like-button"
  >
    Like
  </button>
</DocumentFragment>
`;
